modules/hitlnext/src/views/full/app/components/Tags.tsx

```diff
@@ -0,0 +1,90 @@
+import { Collapsible, lang } from 'botpress/shared'
+import _ from 'lodash'
+import React, { FC, Fragment, useContext, useState } from 'react'
+import { MenuItem } from '@blueprintjs/core'
+import { MultiSelect } from '@blueprintjs/select'
+
+import { IHandoff } from '../../../../types'
+import { ApiType } from '../../Api'
+import { Context } from '../Store'
+
+interface Props {
+  api: ApiType
+  handoff: IHandoff
+}
+
+const StringMultiSelect = MultiSelect.ofType<string>()
+
+export const Tags: FC<Props> = ({ handoff, api }) => {
+  const { tags, id } = handoff
+  const { state } = useContext(Context)
+
+  const [expanded, setExpanded] = useState(true)
+  const [items, setItems] = useState(_.compact(_.castArray(tags)))
+  const [availableItems, setAvailableItems] = useState(state.config.tags)
+
+  function handleSelect(value, index) {
+    const updated = [...items, value]
+    setItems(updated) // Optimistic update
+    api.updateHandoff(id, { tags: updated }).catch(error => setItems(updated.slice(0, -1)))
+  }
+
+  function handleRemove(value, index) {
+    const updated = _.filter(items, (v, i) => i != index)
+    setItems(updated) // Optimistic update
+    api.updateHandoff(id, { tags: updated }).catch(error => setItems(updated.slice(0, -1)))
```

Comment:
.catch doesn't work, you need to add back the item that that has been removed. 

This is probably some copy/paste art

Reply:
Good catch!